test/python/test_registration.py

```diff
@@ -0,0 +1,179 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2018, IBM.
+#
+# This source code is licensed under the Apache License, Version 2.0 found in
+# the LICENSE.txt file in the root directory of this source tree.
+
+# pylint: disable=invalid-name
+
+"""
+Test the registration and credentials features of the wrapper.
+"""
+import os
+from contextlib import contextmanager
+from tempfile import NamedTemporaryFile
+from unittest.mock import patch
+from unittest import skipIf
+
+import qiskit
+from qiskit import QISKitError
+from qiskit.backends.ibmq import IBMQProvider
+from qiskit.wrapper.credentials import _configrc, _qconfig, discover_credentials
+from qiskit.wrapper.credentials._environ import VARIABLES_MAP
+from .common import QiskitTestCase
+
+
+# TODO: NamedTemporaryFiles do not support name in Windows
+@skipIf(os.name == 'nt', 'Test not supported in Windows')
+class TestWrapperCredentuals(QiskitTestCase):
+    """Wrapper autoregistration and credentials test case."""
+    def test_autoregister_no_credentials(self):
+        """Test register() with no credentials available."""
+        with no_file('Qconfig.py'), no_file(_configrc.DEFAULT_QISKITRC_FILE), no_envs():
+            with self.assertRaises(QISKitError) as cm:
+                qiskit.wrapper.register()
+
+        self.assertIn('No IBMQ credentials found', str(cm.exception))
+
+    def test_store_credentials(self):
+        """Test storing credentials and using them for autoregister."""
+        with no_file('Qconfig.py'), no_envs(), custom_qiskitrc(), mock_ibmq_provider():
+            qiskit.wrapper.store_credentials('QISKITRC_TOKEN', proxies={'http': 'foo'})
+            provider = qiskit.register()
+
+        self.assertEqual(provider._token, 'QISKITRC_TOKEN')
+        self.assertEqual(provider._proxies, {'http': 'foo'})
+
+    def test_store_credentials_overwrite(self):
```

Comment:
I would break this test in several, one testing overwriting fails, another for testing failure implies no data is overwritten.